android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/translate/LanguageSetting.kt

```diff
@@ -42,4 +42,84 @@ enum class LanguageSetting(private val languageSetting: String) {
                 throw IllegalArgumentException("The language setting $languageSetting is not mapped.")
         }
     }
+
+    /**
+     * Helper function to transform a given [LanguageSetting] setting into its boolean counterpart.
+     *
+     * @param categoryToSetFor The [LanguageSetting] type that we would like to determine the
+     * boolean value for. For example, if trying to calculate a boolean 'isAlways',
+     * [categoryToSetFor] would be [LanguageSetting.ALWAYS].
+     *
+     * @return A boolean that corresponds to the language setting. Will return null if not enough
+     * information is present to make a determination.
+     */
+    fun toBoolean(
+        categoryToSetFor: LanguageSetting,
```

Comment:
Might be able to simplify and remove `categoryToSetFor`, need to check against dependent patch more.

Reply:
Opening up for review, use in context: 
```
 val setting = pageLanguage?.let { getLanguageSetting(it) }
            val alwaysTranslateLanguage = setting?.toBoolean(LanguageSetting.ALWAYS)
            val neverTranslateLanguage = setting?.toBoolean(LanguageSetting.NEVER)

```
in #5348 because usually the `this` won't be known ahead of time and it is essentially converting types.